### PR TITLE
Libp2p da topic

### PR DESCRIPTION
**Pull request by bfish713**
Add a way to add a topic to libp2p for sending DA messages over.  Also adds a way to configure a static committee.  We still need to add the integration into the rest of the code to get smaller static committees but the network layer should be able to handle it.

Closes: #1065

### CODE REVIEW

File: testing/src/test_launcher.rs [hunk 1]
```diff
@@ -98,13 +99,20 @@ where
                 TYPES::SignatureKey::from_private(&priv_key)
             })
             .collect();
+        let da_committee_nodes = (0..da_committee_size)
+            .map(|id| {
+                let priv_key = I::generate_test_key(id as u64);
+                TYPES::SignatureKey::from_private(&priv_key)
+            })
+            .collect();
```
Comment (DieracDelta) [inline testing/src/test_launcher.rs:107]: Same question here

**Review by DieracDelta — APPROVED** [review]
LGTM sans one question

File: src/traits/networking/libp2p_network.rs [hunk 2]
```diff
@@ -135,16 +135,24 @@ where
...
             let pubkey = TYPES::SignatureKey::from_private(&privkey);
             all_keys.insert(pubkey);
         }
 
+        for i in 0u64..(da_committee_size as u64) {
+            let privkey = TYPES::SignatureKey::generate_test_key(i);
+            let pubkey = TYPES::SignatureKey::from_private(&privkey);
+            da_keys.insert(pubkey);
+        }
+
```
[the remark this answers is not in the document]
Comment (bfish713) [reply src/traits/networking/libp2p_network.rs:155]: Yeah they have to be the same key, I should combine the loops it's pretty simple

**Review by shenkeyao — APPROVED** [review]
LGTM. Looks like we need to re-run the timed-out CI workflow.

**Review by shenkeyao — APPROVED** [review]
LGTM

File: src/traits/networking/libp2p_network.rs [hunk 3]
```diff
@@ -135,13 +135,18 @@ where
...
+        da_committee_size: usize,
     ) -> Box<dyn Fn(u64) -> Self + 'static> {
         let bootstrap_addrs: PeerInfoVec = Arc::default();
         let mut all_keys = BTreeSet::new();
+        let mut da_keys = BTreeSet::new();
 
         for i in 0u64..(expected_node_count as u64) {
             let privkey = TYPES::SignatureKey::generate_test_key(i);
             let pubkey = TYPES::SignatureKey::from_private(&privkey);
+            if i < da_committee_size as u64 {
```
Comment (DieracDelta) [inline src/traits/networking/libp2p_network.rs:147]: Might be worth asserting that `da_commitee_size < expected_node_count`

File: testing/src/test_launcher.rs [hunk 4]
```diff
@@ -139,14 +140,12 @@ where
...
-        let committee_network_generator =
-            I::committee_network_generator(total_nodes, num_bootstrap_nodes);
+        let network_generator =
+            I::network_generator(total_nodes, num_bootstrap_nodes, da_committee_size);
         Self {
             generator: ResourceGenerators {
-                quorum_network_generator,
-                committee_network_generator,
+                network_generator,
+                // committee_network_generator,
```
Comment (DieracDelta) [inline testing/src/test_launcher.rs:148]: can we delete this line?

File: testing/src/test_runner.rs [hunk 5]
```diff
@@ -164,16 +164,16 @@ where
...
-                    node_id,
-                ));
-            let quorum_network = (self.launcher.generator.quorum_network)(quorum_network_generator);
-            let committee_network =
-                (self.launcher.generator.committee_network)(committee_network_generator);
+            let network_generator = Arc::new((self.launcher.generator.network_generator)(node_id));
+            // let committee_network_generator =
+            //     Arc::new((self.launcher.generator.committee_network_generator)(
+            //         node_id,
+            //     ));
```
Comment (DieracDelta) [inline testing/src/test_runner.rs:173]: can we delete this?

**Review by DieracDelta — APPROVED** [review]
LGTM

**Review by DieracDelta — DISMISSED** [review]

**Review by DieracDelta — APPROVED** [review]

**Pull request merged**